Generals/Code/GameEngine/Source/GameNetwork/ConnectionManager.cpp

```diff
@@ -619,21 +619,102 @@ void ConnectionManager::processChat(NetChatCommandMsg *msg)
 		name = m_connections[playerID]->getUser()->GetName();
 		//DEBUG_LOG(("connection is non-NULL, using %ls", name.str()));
 	}
-	unitext.format(L"[%ls] %ls", name.str(), msg->getText().str());
-//	DEBUG_LOG(("ConnectionManager::processChat - got message from player %d (mask %8.8X), message is %ls", playerID, msg->getPlayerMask(), unitext.str()));
 
 	AsciiString playerName;
 	playerName.format("player%d", msg->getPlayerID());
 	const Player *player = ThePlayerList->findPlayerWithNameKey( TheNameKeyGenerator->nameToKey( playerName ) );
 	if (!player)
 	{
+		unitext.format(L"[%ls] %ls", name.str(), msg->getText().str());
 		TheInGameUI->message(UnicodeString(L"%ls"), unitext.str());
 		return;
 	}
 
-	Bool fromObserver = !player->isPlayerActive();
-	Bool amIObserver = !ThePlayerList->getLocalPlayer()->isPlayerActive();
-	Bool canSeeChat = (amIObserver || !fromObserver) && !TheGameInfo->getConstSlot(playerID)->isMuted();
+	// =============================================================
+	// ConnectionManager::processChat()
+	// Refactored by TheSuperHackers @feature - 31/10/2025
+	// Simplified team chat detection + localizable format via FETCH_OR_SUBSTITUTE_FORMAT()
+	// =============================================================
+
+	static Bool isTeamChat(const Player* sender, UInt32 mask)
+	{
+		Int allies = 0;
+		Int recipients = 0;
+
+		for (Int i = 0; i < MAX_SLOTS; ++i)
+		{
+			if ((1 << i) & mask)
+			{
+				const Player* receiver = ThePlayerList->getConstSlot(i);
+				if (receiver && receiver->isPlayerActive())
+				{
+					recipients++;
+					if (sender->getRelationship(receiver->getDefaultTeam()) == ALLIES)
```

Comment:
`ToggleInGameChat` also uses `|| player==localPlayer` as condition. Is this necessary as well or redundant?